jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleResolver.java

```diff
@@ -525,4 +623,46 @@ static long getJ9ClassFromClass(Class<?> c) {
 			return JITHELPERS.getJ9ClassFromClass64(c);
 		}
 	}
+
+	/**
+	 * Used during the invokehandle bytecode for resolving calls to the polymorphic
+	 * MethodHandle and VarHandle methods. The resolution yields two values, which
+	 * are returned in a two element array. The first array element is a MemberName
+	 * object, which specifies the caller method to be invoked. The second array
+	 * element is a MethodType object if the defining class is a MethodHandle and
+	 * an AccessDescriptor object if the defining class is a VarHandle.
+	 *
+	 * This is only used for the OpenJDK MethodHandles, and an InternalError will
+	 * be thrown if it is used for the OpenJ9 MethodHandles.
+	 *
+	 * @param callerClass the caller class
+	 * @param refKind the reference kind used by the CONSTANT_MethodHandle entries
+	 * @param definingClass the defining class
+	 * @param name contains the method name
+	 * @param type contains the method description
+	 *
+	 * @return a two element array, which contains the resolved values
+	 *
+	 * @throws InternalError if invoked for the OpenJ9 MethodHandles
+	 */
+	@SuppressWarnings("unused")
+	private static final Object linkCallerMethod(Class<?> callerClass, int refKind, Class<?> definingClass, String name, String type) throws Throwable {
+/*[IF OPENJDK_METHODHANDLES]*/
+		VMLangAccess access = VM.getVMLangAccess();
+		MethodType mt = MethodType.fromMethodDescriptorString(type, access.getClassloader(callerClass));
+
+		Object[] result = new Object[2];
+		Object[] appendixResult = new Object[1];
+
+		/* result[0] contains the MemberName object, which specifies the caller method (generated bytecodes). */
+		result[0] = MethodHandleNatives.linkMethod(callerClass, refKind, definingClass, name, mt, appendixResult);
```

Comment:
Does this throw `IllegalAccessException` or `IllegalAccessError` for MHs that fail access checking?  The spec expects the Error version but at least with the OJ9 handles, we java code threw the Exception and we had to convert to the Error

Reply:
`MethodHandleNatives.linkMethod` doesn't throw `IllegalAccessException` or `IllegalAccessError`.

MHs that fail access checking should be handled by `sendResolveMethodHandle`. We have disabled the access checks since we don't have an alternative for `accessCheckArgRetTypes`. After looking at the code recently, I think `sendResolveMethodHandle` (in OpenJ9) should be replaced with [`MethodHandleNatives.linkMethodHandleConstant`](https://github.com/ibmruntimes/openj9-openjdk-jdk15/blob/0757efcf6c12bf62a33b6681d48b6e08253e5241/src/java.base/share/classes/java/lang/invoke/MethodHandleNatives.java#L600) (in OpenJDK) since it handles all the error checking related to resolving a `CONSTANT_MethodHandle` entry. Added replacing `sendResolveMethodHandle` with `MHN.linkMethodHandleConstant` as a TODO in https://github.com/eclipse/openj9/issues/10913.